api/v1/types.go

```diff
@@ -1,36 +1,198 @@
 package v1
 
 import (
+	"context"
+	"strings"
 	"time"
 
-	"github.com/leptonai/gpud/components"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// Component represents an individual component of the system.
+//
+// Each component check is independent of each other.
+// But the underlying implementation may share the same data sources
+// in order to minimize the querying overhead (e.g., nvidia-smi calls).
+//
+// Each component implements its own output format inside the State struct.
+// And recommended to have a consistent name for its HTTP handler.
+// And recommended to define const keys for the State extra information field.
+type Component interface {
```

Comment:
it is reasonable that other than `Component` and `HealthSettable` are moved to this file..`Component` and `HealthSettable` should be moved to original file `components/components.go` or `components/types.go`